llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -387,10 +387,6 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
 
   bool isAlwaysUniform(const Value *V) const override { return false; }
 
-  bool isValidAddrSpaceCast(unsigned FromAS, unsigned ToAS) const override {
```

Comment:
TBH I don't understand the relationship between `BasicTTIImpl` and `TargetTransformInfoImplBase`. I just removes this to stop it from overriding my default implementation in TTIIB.

Reply:
`TTIImplBase` has two derived classes, `BasicTTIImpl` and `NoTTIImpl`. The latter is used when the target doesn't provide its own implementation of `TTIImplBase` (which is usually done by inheriting from `BasicTTIImpl`).

Reply:
Probably anything that doesn't make use of TM/TLI (like this and surrounding methods) should only be in TTIImplBase rather than BasicTTIImpl?